3releases: opam 2.0.10 / opam 2.1.1 / opam-depext 1.2 (#80)
ocaml/platform-blog: opam-2-0-10-2-1-1-depext.md

title: "opam releases: 2.0.10, 2.1.1, & opam depext 1.2!"
authors: [
  "David Allsopp - OcamlLabs {"mailto:David.Allsopp(à)cl.cam.ac.uk"}"
  "Raja Boujbel - OCamlPro" {"mailto:raja.boujbel(à)ocamlpro.com"}
  "Louis Gesbert - OCamlPro" {"mailto:louis.gesbert(à)ocamlpro.com"}
]

> date: "2021-11-10"

Author: ```suggestion
date: "2021-11-15"
```

## opam-depext 1.2

Previous versions of opam-depext were made unavailable when using opam 2.1, since depext handling is done directly by opam 2.1 and later. This is still the recommended way, but this left workflows which wanted to maintain compatibility with opam 2.0 without a single command to install depexts. You can now run `OPAMCLI=2.0 opam depext -y package1 package2` and expect this to work correctly with any version of opam 2. If you don't specify `OPAMCLI=2.0` then the plugin will remind you that you should be using the integrated depext support! Calling `opam depext` this way with opam 2.1 and later still exposes the same double-solving problem that opam 2.0 has, but if for some reason the solver returns a different solution at `opam install` then the correct depexts would be installed.

For opam 2.0, some useful depext handling changes are back-ported from opam 2.1.x to the plugin:
With opam 2.0:

> * yum-based distributions: force not to upgrade
> * Archlinux: always upgrade all the installed packages when installing a new package

Reviewer: ```suggestion
* yum-based distributions: force not to upgrade ([#137](https://github.com/ocaml-opam/opam-depext/pull/137))
* Archlinux: always upgrade all the installed packages when installing a new package ([#138](https://github.com/ocaml-opam/opam-depext/pull/138))
```

* Run the sandbox check in a temporary directory ([#4783](https://github.com/ocaml/opam/issues/4783))

Integrated depext support has a few updates:
* Homebrew now has support for casks and full-names ([#4800](https://github.com/ocaml/opam/issues/4800))
* Archlinux now handles virtual package detection ([#4833](https://github.com/ocaml/opam/pull/4833), partially addressing [#4759](https://github.com/ocaml/opam/issues/4759))
* Disable the detection of available packages on RHEL-based distributions.
  This fixes an issue on RHEL-based distributions where yum list used to detect
  available and installed packages would wait for user input without showing
  any output and/or fail in some cases ([#4791](https://github.com/ocaml/opam/pull/4791))
  
And finally two regressions have been dealt with:
* Regression: avoid calling `Unix.environment` on load (as a toplevel expression). This regression affected opam's libraries, rather than the binary itself ([#4789](https://github.com/ocaml/opam/pull/4789))
* Regression: handle empty environment variable updates ([#4840](https://github.com/ocaml/opam/pull/4840))

> A few issues with the compilation of opam from sources have been fixed as well:
> ##`make build`
> works more reliably he the cold target in presence of an older
> * The vendored build of ocaml-mccs on mingw-w64 with g++ 11.2 now works
> * dose3 download URL has been amended, since Inria's GForge is no more

Reviewer: This sort of change aren’t really useful for users so we can slime down the post here instead:
```suggestion
A few issues with the compilation of opam from sources have been fixed as well (e.g. mingw-w64 with g++ 11.2 now works)
```
